test/e2e/integration_test.go

```diff
@@ -0,0 +1,69 @@
+package e2e
+
+import (
+	. "github.com/onsi/ginkgo/v2"
+	bmov1alpha1 "github.com/metal3-io/baremetal-operator/apis/metal3.io/v1alpha1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/cluster-api/test/framework"
+)
+
+var _ = Describe("When testing integration [integration]", func() {
+	It("CI Test Provision", func() {
+		numberOfWorkers       = int(*e2eConfig.GetInt32PtrVariable("WORKER_MACHINE_COUNT"))
+		numberOfControlplane = int(*e2eConfig.GetInt32PtrVariable("CONTROL_PLANE_MACHINE_COUNT"))
+		k8sVersion := e2eConfig.GetVariable("KUBERNETES_VERSION")
+		By("Provision Workload cluster")
+		targetCluster, result := createTargetCluster(k8sVersion)
+		By("Pivot objects to target cluster")
+		pivoting(ctx, func() PivotingInput {
+			return PivotingInput{
+				E2EConfig:             e2eConfig,
+				BootstrapClusterProxy: bootstrapClusterProxy,
+				TargetCluster:         targetCluster,
+				SpecName:              specName,
+				ClusterName:           clusterName,
+				Namespace:             namespace,
+				ArtifactFolder:        artifactFolder,
+				ClusterctlConfigPath:  clusterctlConfigPath,
+			}
+		})
+		By("Repivot everything back to the source cluster")
+		rePivoting(ctx, func() RePivotingInput {
+			return RePivotingInput{
+				E2EConfig:             e2eConfig,
+				BootstrapClusterProxy: bootstrapClusterProxy,
+				TargetCluster:         targetCluster,
+				SpecName:              specName,
+				ClusterName:           clusterName,
+				Namespace:             namespace,
+				ArtifactFolder:        artifactFolder,
+				ClusterctlConfigPath:  clusterctlConfigPath,
+			}
+		})
+		By("Deplovisioning target cluster")
+		bootstrapClient := bootstrapClusterProxy.GetClient()
+		intervals := e2eConfig.GetIntervals(specName, "wait-deprovision-cluster")
+		// Capi DeleteClusterAndWait() has a check to make sure all resources managed by CAPI
+		// is gone after Cluster deletion. This check will fails in CAPM3 as all BMH-related 
```

Comment:
Maybe I can help clarify a bit. We label the BMO CRDs for pivot, so that CAPI will take care of moving the objects. Because of this, the e2e framework will also expect them to disappear when deleting the cluster since they are part of the owner chain. So the point of the comment is that we cannot rely on this "delete cluster and wait" method. Instead we just delete the cluster and then separately wait for the relevant resources.